src/main/java/seedu/address/logic/commands/AddIngredientCommand.java

```diff
@@ -67,7 +67,8 @@ private boolean isDuplicateIngredient(Map<Integer, Product> catalogue, Ingredien
                 .filter(product -> product instanceof Ingredient) // Filter only Ingredient objects
                 .map(product -> (Ingredient) product) // Cast to Ingredient
                 .anyMatch(existingIngredient ->
-                        existingIngredient.getName().equalsIgnoreCase(newIngredient.getName()));
+                        existingIngredient.getName().equalsIgnoreCase(newIngredient.getName()) &&
+                                Double.compare(existingIngredient.getCost(), newIngredient.getCost()) == 0);
```

Comment:
Bro you shouldnt add this && cost cus it allows adding two ingredients with same name but diff cost